composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MapView.kt

```diff
@@ -9,40 +9,41 @@ import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material.OutlinedTextField
 import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
-import androidx.compose.runtime.rememberCoroutineScope
-import androidx.compose.runtime.setValue
+import androidx.compose.material.TextFieldDefaults
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
 import androidx.compose.ui.unit.dp
 import com.overdrive.cruiser.endpoints.SearchBoxFetcher
 import com.overdrive.cruiser.models.Coordinate
 import com.overdrive.cruiser.models.Spot
-import com.overdrive.cruiser.models.mapbox.Suggestion
 import kotlinx.coroutines.launch
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.runtime.*
 
 /**
  * A composable that displays a map with a search box and suggestions.
  *
  * @param spots The spots to display on the map.
  */
 @Composable
-fun MapView(spots: List<Spot>) {
+fun MapView(spots: List<Spot>, mapViewModel: MapViewModel) {
     val scope = rememberCoroutineScope()
     val suggestionGenerator = remember { SearchBoxFetcher() }
+    val focusManager = LocalFocusManager.current
 
-    var query by remember { mutableStateOf("") }
-    var suggestions by remember { mutableStateOf(emptyList<Suggestion>()) }
-    var currentLocation by remember { mutableStateOf(Coordinate.DEFAULT) }
+    val query by mapViewModel.query.collectAsState("")
+    val suggestions by mapViewModel.suggestions.collectAsState(emptyList())
+    val currentLocation by mapViewModel.currentLocation.collectAsState(Coordinate.DEFAULT)
```

Comment:
Should these not still be in `remember { }` blocks?

Reply:
probably will have to look back to this later, remember blocks don't seem to work well with collectAsState.